minimal.js

```diff
@@ -128,20 +141,14 @@ module.exports = function (dirname, keys, opts) {
       return cb(err)
     }
 
-    queue(msg, function (err) {
+    queue(msg, function (err, message) {
```

Comment:
I think there's a behaviour change here. Your `queue` calls back with _every_ callback passed to it.
AsyncWrite runs the callback when:
1. (buffer is full)
   - a), and (not already writing)
   - b)  and (already writing) AND **(is was the latest callback added)**
2. (buffer is not full)

Path 1 (b) looks like it basically swallows all but the last callback from running  once the buffer was already full

Reply:
https://github.com/dominictarr/async-write/blob/master/index.js#L36

Reply:
I _think_ that could be the bug which was returning undefined after you top 1000 messages

Reply:
Yeah, it looks like while Async-Write is writing it drops all callbacks except the most recent one added. So if your buffer is full after 1000 messages and you supply 2000, it'll drop callbacks 1001-1999.

The weird thing is that the bug wasn't that the callbacks weren't running, it's that they *were* running but `msg === undefined`. I don't really understand why.

Reply:
:+1: